ludwig/datasets/dataset_config.py

```diff
@@ -29,6 +29,9 @@ class DatasetConfig:
     # The readable description of the dataset
     description: str = ""
 
+    # The learning task(s) associated with this dataset.
+    task: str = ""
```

Comment:
Recommend calling it tasks, even when there is only one item.
```suggestion
    tasks: Union[str, List[str]] = field(default_factory=list)
```


Also IMO it would be cleaner to make this a `Union[str, List[str]]`, similar to `download_urls`. Then we could do:

```
task: Regression
```
or
```
task:
  - Regression
  - Classification
```